torch_frame/data/multi_nested_tensor.py

```diff
@@ -350,13 +350,34 @@ def dtype(self) -> torch.dtype:
 
     # Static methods ##########################################################
     @staticmethod
-    def stack(xs: List['MultiNestedTensor'],
-              dim: int = 0) -> 'MultiNestedTensor':
-        # TODO: To be implemented.
-        if len(xs) == 1:
-            return xs[0]
+    def cat(xs: List['MultiNestedTensor'],
+            dim: int = 0) -> 'MultiNestedTensor':
+        if dim == 0 or dim == -3:
+            num_rows = sum(x.num_rows for x in xs)
+            num_cols = xs[0].num_cols
+            for x in xs:
+                if x.num_cols != num_cols:
+                    raise RuntimeError(
+                        "num_cols must be the same across a list of input "
+                        "multi nested tensors.")
+            values = torch.cat([x.values for x in xs], dim=0)
+            accum_offset = 0
+            offset = []
+            for x in xs[:-1]:
+                offset.append(x.offset[:-1] + accum_offset)
+                accum_offset = accum_offset + x.offset[-1]
+            offset.append(xs[-1].offset + accum_offset)
```

Comment:
This operation is O(num_cols * num_rows) and then you then need to do `torch.cat` which is also quadratic. It would be inefficient. 
Can we do something like 
```
offset = torch.zeros(num_rows * num_cols + 1)
for i in range(1, num_rows * num_cols + 1):
    offset[i] = ...
```
So we can half the runtime?

Reply:
torch.cat is not quadratic. I think this is ok.

Reply:
Hmm, I think this usage of torch.cat is pretty standard to me.

Reply:
Ok, I managed to make `cat` more memory efficient: https://github.com/pyg-team/pytorch-frame/pull/161/commits/aff3240c9b3416d26edb4a445d868f37c83a3bbc